compilers/concrete-compiler/compiler/lib/Support/CompilationFeedback.cpp

```diff
@@ -1,19 +1,69 @@
 // Part of the Concrete Compiler Project, under the BSD3 License with Zama
+
 // Exceptions. See
 // https://github.com/zama-ai/concrete-compiler-internal/blob/main/LICENSE.txt
 // for license information.
 
 #include <cassert>
 #include <fstream>
+#include <vector>
 
+#include "concrete-protocol.capnp.h"
 #include "concretelang/Support/CompilationFeedback.h"
 
 using concretelang::protocol::Message;
 
 namespace mlir {
 namespace concretelang {
 
-void CompilationFeedback::fillFromProgramInfo(
+void CircuitCompilationFeedback::fillFromCircuitInfo(
+    concreteprotocol::CircuitInfo::Reader circuitInfo) {
+  auto computeGateSize =
+      [&](const Message<concreteprotocol::GateInfo> &gateInfo) {
+        unsigned int nElements = 1;
+        // TODO: CHANGE THAT ITS WRONG
```

Comment:
?

Reply:
It's a leftover from a while ago. The code below looks legit to me anyway.

Reply:
yes I think too, can you blame the TODO and ask to the author then just remove that comment if ok. Then you can resolve this conversation